nomos-services/network/src/backends/mixnet/mod.rs

```diff
@@ -17,47 +57,156 @@ impl NetworkBackend for MixnetNetworkBackend {
     type EventKind = EventKind;
     type NetworkEvent = NetworkEvent;
 
-    fn new(_config: Self::Settings, _overwatch_handle: OverwatchHandle) -> Self {
+    fn new(config: Self::Settings, overwatch_handle: OverwatchHandle) -> Self {
         // TODO: One important task that should be spawned is
         // subscribing NewEntropy events that will be emitted from the consensus service soon.
         // so that new topology can be built internally.
         // In the mixnet spec, the robustness layer is responsible for this task.
         // We can implement the robustness layer in the mixnet-specific crate,
         // that we're going to define at the root of the project.
-        todo!()
+
+        let (libp2p_commands_tx, libp2p_commands_rx) = tokio::sync::mpsc::channel(BUFFER_SIZE);
+        let (libp2p_events_tx, libp2p_events_rx) = tokio::sync::broadcast::channel(BUFFER_SIZE);
+        let (events_tx, _) = tokio::sync::broadcast::channel(BUFFER_SIZE);
+
+        let mut swarm_handler = SwarmHandler::new(
+            &config.libp2p_config,
+            libp2p_commands_tx.clone(),
+            libp2p_commands_rx,
+            libp2p_events_tx.clone(),
+        );
+
+        // Run mixnode
+        let (mixnode, packet_queue) = MixNode::new(config.mixnode_config).unwrap();
+        let libp2p_cmd_tx = libp2p_commands_tx.clone();
+        let queue = packet_queue.clone();
+        overwatch_handle.runtime().spawn(async move {
```

Comment:
Shall we collect all of the `JoinHandle`s, so that when we can shut down gracefully, we can wait for all spawned tasks to be finished?

Reply:
Good point. I think this should be done as a separate PR for all other codes as well as this.